src/main/java/seedu/address/logic/commands/ListClientCommand.java

```diff
@@ -33,15 +37,15 @@ public ListClientCommand(DateKeyword dateKeyword) {
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        if (this.object == null) {
-            model.updateFilteredClientList(PREDICATE_SHOW_ALL_CLIENTS);
-        }
+        Predicate<Client> pred = PREDICATE_SHOW_ALL_CLIENTS;
+
         if (this.object instanceof DateKeyword) {
-            model.updateFilteredClientList(client -> client.isBirthdayInPeriod((DateKeyword) object));
+            pred = client -> client.isBirthdayInPeriod((DateKeyword) object);
         }
         if (this.object instanceof Product) {
```

Comment:
should probably do else if since it is one of two

Reply:
coz still got the condition where it is no command, so I dun think can use `else` here

Reply:
as in` else if` not `else`, since they belong in the same group, but just a nitpick

Reply:
ah i see, coz my worry is the guard clause.

Reply:
I feel like this i should leave it as it is for guard clause

Reply:
can also

Reply:
just a bit of a nitpick tbh, functionality doesnt affect